### PR TITLE
feat(central): show progress when uploading data csv files

### PR DESCRIPTION
**Pull request by harmbrugge**
#3059 

What are the main changes you did:
- The CsvApi will progress a post request async as a Task
- Updated the front-end accordingly.

how to test:
- Upload any data from a csv file into a table.

todo:
- [x] updated docs in case of new feature
- [x] added/updated tests

### CODE REVIEW

**Review by mswertz — CHANGES_REQUESTED** [review]
Add junit test to get coverage over 80% (actually sonar should have failed).
I think if you would add a one liner for the ?async=true in WebApiSmokeTests you would cover all java code.

You can see here what code is not covered
https://sonarcloud.io/component_measures?id=molgenis_molgenis-emx2&pullRequest=4115&metric=new_coverage&view=list

**Review by mswertz — APPROVED** [review]

**Comment by davidruvolo51** [comment]
I performed a functional review of the csv import. I noticed a couple of issues. 

## Import does not timeout or show errors

I created a test dataset for the pet store and tried to import it. While I see the nice import display, the progress does not update. I don't see any errors or updates regarding the import. In the dev console, I see that the graphql requests are executing fine, but the response is the same:

```
{
  "data" : {
    "_tasks" : [
      {
        "id" : "e4ae8c38-ad8e-4626-b55c-b61e2a8aa852",
        "description" : "Importing rows into Pet for 321604ms",
        "status" : "RUNNING",
        "subTasks" : [ ]
      }
    ]
  }
}
```

To reproduce this issue -

1. Sign in as admin
2. Go to up/download page in the pet store
3. Import this csv file [Pet.csv](https://github.com/user-attachments/files/16530459/Pet.csv)

## The progress indicator isn't shown when importing a `molgenis.csv` file

When importing a `molgenis.csv` file, the progress indicator is not display.

1. Create a new schema
2. Import the following csv file: [molgenis.csv](https://github.com/user-attachments/files/16530527/molgenis.csv)

File: apps/updownload/src/components/Import.vue [hunk 1]
```diff
@@ -179,8 +183,14 @@ export default {
           fetch(url, options)
             .then((response) => {
               if (response.ok) {
-                response.text().then((successText) => {
-                  this.success = successText;
+                response.json().then((response) => {
```
Comment (davidruvolo51) [inline apps/updownload/src/components/Import.vue:186]: More of a preference really, but in the `response.ok` check can also return `response.json()` and pass it to another `then`.

```
fetch(...)
.then((response) => {
    if (response.ok) {
       return response.json()
    }
  ...
})
.then((result) => {
   // process the response here
})
.catch((err) => {
   // handle error here
})
```

**Review by davidruvolo51 — APPROVED** [review]
Looks good! I have one comment regarding the structure of the fetch. Otherwise, everything works fine.

File: apps/updownload/src/components/Import.vue [hunk 2]
```diff
@@ -179,8 +183,14 @@ export default {
...
+                  if (response.id) {
+                    // if identifier is present it's a task
```
Comment (chinook25) [inline apps/updownload/src/components/Import.vue:188]: Try to not add comments like these, if you really want to make the code more clear use functions/variable names. Here you could do `if (isTask(response))`.

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/CsvApi.java [hunk 3]
```diff
@@ -56,17 +57,24 @@ static String mergeMetadata(Request request, Response response) {
     String fileName = request.headers("fileName");
     boolean fileNameMatchesTable = getSchema(request).getTableNames().contains(fileName);
 
-    if (fileNameMatchesTable) {
-      // so we assume it isn't meta data
-      int count = MolgenisWebservice.getTableById(request, fileName).save(getRowList(request));
-      response.status(200);
-      response.type(ACCEPT_CSV);
-      return "imported number of rows: " + count;
+    if (fileNameMatchesTable) { // so we assume it isn't meta data
```
Comment (chinook25) [inline backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/CsvApi.java:60]: metadata is one word

**Pull request merged**